internal/message/subtrie.go

```diff
@@ -54,6 +55,13 @@ func NewTrie() *Trie {
 	}
 }
 
+// NewTrieWithTopicMatchingType creates a new matcher with specified matching type for the subscriptions.
+func NewTrieWithTopicMatchingType(topicMatchingType string) *Trie {
+	t := NewTrie()
+	t.topicMatchingType = topicMatchingType
```

Comment:
I would simplify and decouple this function from the config itself. Something like this should be good

```
// newTrie creates a new trie without a lookup function
func newTrie() *Trie {
   return &Trie{
      root: &node{
         subs:     newSubscribers(),
         children: make(map[uint32]*node),
      },
   }
}
```

```
// NewTrie creates a new subscriptions matcher using standard emitter strategy.
func NewTrie() *Trie {
   t :=newTrie()
   t.lookup = t.lookupEmitter
   return t
}
```

```
// NewTrieMQTT creates a new subscriptions matcher using standard MQTT strategy.
func NewTrieMQTT() *Trie {
   t := newTrie()
   t.lookup = t.lookupMqtt
   return t
}
```